### PR TITLE
Fixes #16807: rl add cfengine_core to libs, which declares global variables

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/16807

can be merged unless some categories are not necessary

### CODE REVIEW

**Comment by peckpeck** [comment]
one commit is still a wip here

**Comment by gpoblon** [comment]
PR rebased

**Comment by gpoblon** [comment]
Commit modified

File: rudder-lang/libs/cfengine_core.rl [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+declare const.r
+declare const.t
+
+# `connection` global variables
+declare connection.key
+declare connection.ip
+declare connection.hostname
+
+# `match` global variables
+declare match.0
```
Comment (amousset) [inline rudder-lang/libs/cfengine_core.rl]: technically it can be any int value as it is used for regex captures

Comment (gpoblon) [reply rudder-lang/libs/cfengine_core.rl]: Absolutely, I forgot about this one

**Comment by gpoblon** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**